commit 81562a0b312c08dc0ff95c47fce64c9837bfc39f (refs/changes/53/753/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-10-20T14:16:02-07:00 (3 years ago)
    
    PUBAPI-1338 use cueball for talking to SDC services
    Reviewed by: Marsell Kukuljevic <marsell@joyent.com>

diff --git a/lib/app.js b/lib/app.js
index 06b9ab4..319c2c4 100644
--- a/lib/app.js
+++ b/lib/app.js
@@ -30,6 +30,7 @@ var UFDS = require('ufds');
 var keyapi = require('keyapi');
 var mahi = require('mahi');
 var cueball = require('cueball');
+var kang = require('kang');
 
 var account = require('./account');
 var analytics = require('./analytics');
@@ -205,8 +206,14 @@ function createClients(options, callback) {
     options.imgapi.log = options.log.child({ component: 'imgapi' });
     options.papi.log   = options.log.child({ component: 'papi' });
     options.fwapi.log  = options.log.child({ component: 'fwapi' });
+    options.cnapi.log  = options.log.child({ component: 'cnapi' });
 
     options.fwapi.agent = agent;
+    options.cnapi.agent = agent;
+    options.papi.agent = agent;
+    options.napi.agent = agent;
+    options.vmapi.agent = agent;
+    options.imgapi.agent = agent;
 
     if (options.mahi) {
         options.mahi.log = options.log.child({ component: 'mahi' });
@@ -216,6 +223,7 @@ function createClients(options, callback) {
 
     if (options.cns) {
         options.cns.log = options.log.child({ component: 'cns' });
+        options.cns.agent = agent;
     }
 
     // On this case, we'll point pkg and ufds to the same UFDS server,
@@ -383,6 +391,18 @@ module.exports = {
             }
         }
 
+        var kangOpts = cueball.poolMonitor.toKangOptions();
+        kangOpts.port = config.port + 1010;
+        /*
+         * Note that we can't use kang.knStartServer here, as kang's restify
+         * version does not match ours and the two will clobber each other.
+         */
+        var kangServer = restify.createServer({ serverName: 'Kang' });
+        kangServer.get(new RegExp('.*'), kang.knRestifyHandler(kangOpts));
+        kangServer.listen(kangOpts.port, '127.0.0.1', function () {
+            log.info('cueball kang monitor started on port %d', kangOpts.port);
+        });
+
         return createClients(config, function (error, clients) {
             if (error) {
                 log.error({err: error}, 'Create clients error');
diff --git a/package.json b/package.json
index a22fb69..1d8fc06 100644
--- a/package.json
+++ b/package.json
@@ -21,7 +21,8 @@
         "nopt": "2.0.0",
         "restify": "git+https://github.com/mcavage/node-restify.git#0d7b4ba",
         "bunyan": "0.22.1",
-        "cueball": "1.1.1",
+        "cueball": "1.1.2",
+        "kang": "1.1.0",
         "sdc-clients": "9.4.0",
         "ufds": "1.1.3",
         "semver": "2.2.1",
diff --git a/sapi_manifests/cloudapi/template b/sapi_manifests/cloudapi/template
index 516ca87..34fd725 100644
--- a/sapi_manifests/cloudapi/template
+++ b/sapi_manifests/cloudapi/template
@@ -66,11 +66,21 @@
     "cueballHttpAgent": {
         "resolvers": ["{{{BINDER_SERVICE}}}"],
         "initialDomains": [
+            "{{{vmapi_domain}}}",
+            "{{{cnapi_domain}}}",
+            "{{{napi_domain}}}",
+            "{{{fwapi_domain}}}",
+            "{{{imgapi_domain}}}",
+{{#CNS_SERVICE}}
+            "{{{cns_domain}}}",
+{{/CNS_SERVICE}}
+{{#MAHI_SERVICE}}
             "{{{mahi_domain}}}",
-            "{{{fwapi_domain}}}"
+{{/MAHI_SERVICE}}
+            "{{{papi_domain}}}"
         ],
         "spares": 4,
-        "maximum": 10,
+        "maximum": 20,
         "recovery": {
             "default": {
                 "timeout": 2000,
